ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/hateoas/HateoasMapperImpl.java

```diff
@@ -0,0 +1,189 @@
+                continue;
+            }
+            FunctionDefinitionNode resource = (FunctionDefinitionNode) node;
+            Optional<ApiResponses> responses = getApiResponsesForResource(resource, paths);
+            if (responses.isEmpty()) {
+                continue;
+            }
+            setOpenApiLinksInApiResponse(serviceId, resource, responses.get());
+        }
+    }
+
+    private Optional<ApiResponses> getApiResponsesForResource(FunctionDefinitionNode resource, Paths paths) {
+        String resourcePath = MapperCommonUtils.unescapeIdentifier(generateRelativePath(resource));
+        if (!paths.containsKey(resourcePath)) {
+            return Optional.empty();
+        }
+        PathItem openApiResource = paths.get(resourcePath);
+        String httpMethod = resource.functionName().toString().trim();
+        ApiResponses responses = null;
+        switch (httpMethod.trim().toUpperCase(Locale.ENGLISH)) {
+            case Constants.GET -> {
+                responses = openApiResource.getGet().getResponses();
+            }
+            case Constants.PUT -> {
+                responses = openApiResource.getPut().getResponses();
+            }
+            case Constants.POST -> {
+                responses = openApiResource.getPost().getResponses();
+            }
+            case Constants.DELETE -> {
+                responses = openApiResource.getDelete().getResponses();
+            }
+            case Constants.OPTIONS -> {
+                responses = openApiResource.getOptions().getResponses();
+            }
+            case Constants.PATCH -> {
+                responses = openApiResource.getPatch().getResponses();
+            }
+            case Constants.HEAD -> {
+                responses = openApiResource.getHead().getResponses();
+            }
+            default -> { }
+        }
+        return Optional.ofNullable(responses);
+    }
+
+    private void setOpenApiLinksInApiResponse(int serviceId, FunctionDefinitionNode resource,
+                                              ApiResponses apiResponses) {
+        Map<String, Link> swaggerLinks = mapHateoasLinksToOpenApiLinks(packageId, serviceId, resource);
+        if (swaggerLinks.isEmpty()) {
+            return;
+        }
+        for (Map.Entry<String, ApiResponse> entry : apiResponses.entrySet()) {
+            int statusCode = Integer.parseInt(entry.getKey());
+            if (statusCode >= 200 && statusCode < 300) {
+                entry.getValue().setLinks(swaggerLinks);
+            }
+        }
+    }
+
+    private Map<String, Link> mapHateoasLinksToOpenApiLinks(String packageId, int serviceId,
+                                                            FunctionDefinitionNode resourceFunction) {
+        Optional<String> linkedTo = getResourceConfigAnnotation(resourceFunction)
+                .flatMap(resourceConfig -> getValueForAnnotationFields(resourceConfig, BALLERINA_LINKEDTO_KEYWORD));
```

Comment:
Can we try to get the ExpressionNode here rather than the string value of that node. IIRC syntax tree already parses the values and create them with proper structure. If we use that type, then I think we do not need to write our own parser logic

Reply:
This work will be done as an improvement [#5978](https://github.com/ballerina-platform/ballerina-library/issues/5978)